File: manager-web-ui/src/api/apiService.js
import { HttpClient } from "@/utils/http"
import { ApiPaths } from "@/utils/pathConstants"

const getApis = (pageNum, pageSize) => {
  const params = { pageNum: pageNum, pageSize: pageSize }
  return HttpClient.get(ApiPaths.amApi, { params: params })
}

const createApi = (apiDto) => {
  return HttpClient.post(ApiPaths.amApi, apiDto)
}

const publishApi = (apiId) => {
  return HttpClient.post(`${ApiPaths.amApiPublish}/${apiId}`)
}

const offlineApi = (apiId) => {
  return HttpClient.post(`${ApiPaths.amApiOffline}/${apiId}`)
}

const ApiService = {
  getApis, publishApi, offlineApi, createApi
}

export { ApiService }

File: manager/src/main/java/zk/fornax/manager/service/ApiService.java
package zk.fornax.manager.service;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import zk.fornax.manager.bean.PageData;
import zk.fornax.manager.bean.Role;
import zk.fornax.manager.bean.dto.ApiDto;
import zk.fornax.manager.bean.po.ApiEntity;
import zk.fornax.manager.security.HasRole;

public interface ApiService {

    @HasRole(Role.NORMAL_USER)
    Mono<ApiEntity> createApi(ApiDto apiDto);

    @HasRole({ Role.SYSTEM_ADMIN, Role.NORMAL_USER })
    Mono<PageData<ApiEntity>> listApis(int pageNum, int pageSize);

    Mono<ApiEntity> publishApi(String apiId);

    Mono<ApiEntity> offlineApi(String apiId);

    Flux<ApiEntity> listChangedApisAfterLastModifiedTime(long timeMillis);

}
